pkg/install/fixcloudconfig.go

```diff
@@ -0,0 +1,64 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+)
+
+func (i *Installer) fixCloudConfig(ctx context.Context) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		cm, err := i.kubernetescli.CoreV1().ConfigMaps("openshift-config").Get("cloud-provider-config", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		var config map[string]interface{}
+		err = yaml.Unmarshal([]byte(cm.Data["config"]), &config)
+		if err != nil {
+			return err
+		}
+
+		if config["aadClientId"].(string) != "" {
+			i.log.Info("skip fixCloudConfig")
+			return nil
+		}
+
+		s, err := i.kubernetescli.CoreV1().Secrets("kube-system").Get("azure-cloud-provider", metav1.GetOptions{})
+		switch {
+		case errors.IsNotFound(err):
+			i.log.Error("azure-cloud-provider secret not found.")
+		case err != nil:
+			return err
+		}
```

Comment:
can pull this outside the RetryOnConflict.
NotFound presumably should be fatal - missing a return here - why not just do `if err != nil { return err }` ?